### ISSUE TITLE
Allow Manual Editing of Dictionary

### ISSUE THREAD

**Issue by Firefairy** [dslul/openboard#368]
I have been using OpenBoard long enough that it has learned my habits and wording, but this comes with the caveat that it is also learning my frequently-repeated mistakes that I usually don't notice until after I hit send. It is now suggesting several of these as if they were correct. I would like to be able to manually remove these entries from the dictionary, and possibly to manually add idiosyncratic words I use, rather than having to hope I type them correctly when I use them.

**Comment by Mrnofish** [comment]
Have you tried

Settings > System > Languages & Input > On-Screen Keyboard > OpenBoard > Text Correction > Personal Dictionary

?

**Comment by Firefairy** [comment]
> Have you tried
> Settings > System > Languages & Input > On-Screen Keyboard > OpenBoard > Text Correction > Personal Dictionary

Yes. That only has the words I manually saved, not the ones OpenBoard has learned as I type, so I can't take out or block my common errors that I tend not to notice until later, such as "tbe".

**Comment by Firefairy** [comment]
When I say "dictionary", I don't mean the one I can manually save to and that is kept in the system as a whole. I mean the one OpenBoard is using to track words it has picked up from my typing. As an example, it now knows that I sometimes say " woodjit " (a term for a nonsoecified object ", and suggests it if I get as far as "woodj". Thiz is desirable. On the other hand, it also suggest some of the mistakes I tend to complete without noticing, like the aforementioned "tbe" and suggests them as well.

**Comment by Isaacson33** [comment]
I'm finding the same problem - although noticing this request has garnered no interest for over a year, maybe we're the only ones!

I just uninstall and re-install Openboard every month or so just to clear out typing mistakes.

I'd have thought it would be better if the application only learnt frequently used words that were also in the chosen dictionaries, but I'm no coder, maybe that's really difficult.

Just thought I'd share my solution seeing as a fix doesn't seem to be round any corner.

**Comment by Mrnofish** [comment]
> I just uninstall and re-install Openboard every month or so just to clear out typing mistakes.

Wouldn't clearing data achieve the same without the need to reinstall the app?

> I'm finding the same problem - although noticing this request has garnered no interest for over a year, maybe we're the only ones!

I wouldn't assume as much, development on OpenBoard just seems slow and unresponsive, and I am skeptical there is much dogfooding either.

**Comment by Isaacson33** [comment]
> Wouldn't clearing data achieve the same without the need to reinstall the app?

Probably, yes, thanks. I feel a bit daft for not having thought of that.

> I wouldn't assume as much, development on OpenBoard just seems slow and unresponsive

Yes, wish I could help, but unless they need a tree felling, I'm not of much use. Just had a notification as I'm typing that 1.4.5 is available on F-Droid. We'll see what that brings...

**Comment by tdbe** [comment]
I genuinely honestly want to understand the thought process of why devs would release a keyboard where it learns your mistakes but does not expose the list of learned words to reverse them. How can you not imagine that this would be essential?

I really don't wish to be rude but it's baffling to consider the feature RELEASABLE when it will with 100% certainty also do exactly the OPPOSITE of what you need it for.

The keyboard's been inserting "dont"'s in my texts for over a year now. Good job guys. 

------------------------

It would really take just a couple hours for a dev familiar with this project, to expose the learned words dict in settings. You already have the dict, and you already have searchable fields for the custom dict. I'll buy you a coffee, come on.

**Comment by tdbe** [comment]
I added entries for "don't" - shortcut "dont" in the manual dictionary, but it doesn't take priority over the internally auto-learned "dont" 👍gg

**Comment by nach0** [comment]
I face the same problem and would also like to be able to delete the words that OpenBoard has learned.
I believe this is a duplicate of issue #106, so I encourage you all to give it a thumbs up to boost its visibility

**Comment by Mrnofish** [comment]
> I encourage you all to give it a thumbs up to boost its visibility

Sure, why not? Though note there hasn't been a meaningful amount of code contributions since last August, so I think we're SOL.

**Comment by SilverPuppy** [comment]
> I'm finding the same problem - although noticing this request has garnered no interest for over a year, maybe we're the only ones!
> 
> I just uninstall and re-install Openboard every month or so just to clear out typing mistakes.
> 
> I'd have thought it would be better if the application only learnt frequently used words that were also in the chosen dictionaries, but I'm no coder, maybe that's really difficult.
> 
> Just thought I'd share my solution seeing as a fix doesn't seem to be round any corner.

You are NOT the only ones. I'm completely frustrated with the inability to stop it from suggesting my mistakes other than clearing the whole database, which I'm not willing to do. I wish I had the brains to just fix it myself. I agree with @tdbe that this really shouldn't be that difficult, but also, dev interest is clearly lacking.

**Comment by Mrnofish** [comment]
Development has been dead in the water for several years now.

There is a fork that is being actively developed, it's called HeliBoard.

**Comment by SilverPuppy** [comment]
> Development has been dead in the water for several years now.
> 
> There is a fork that is being actively developed, it's called HeliBoard.

Sorry to hear that. But, testing HeliBoard is going well, and a remark in the release notes suggests it may have recently received a similar feature. I haven't had it learn mistakes yet to be sure. 😆 EDIT: Sure enough, long press spawns a trash icon. Haven't tried it on a mistake, though one slipped in already. 😅
